homeassistant/components/recorder/backup.py

```diff
@@ -0,0 +1,23 @@
+"""Backup platform for the Recorder integration."""
+from logging import getLogger
+
+from homeassistant.core import HomeAssistant
+
+from . import Recorder
+from .const import DATA_INSTANCE
+
+_LOGGER = getLogger(__name__)
+
+
+async def async_pre_backup(hass: HomeAssistant) -> None:
+    """Perform operations before a backup starts."""
+    _LOGGER.info("Backup start notification, locking database for writes")
+    instance: Recorder = hass.data[DATA_INSTANCE]
+    await instance.lock_database()
+
+
+async def async_post_backup(hass: HomeAssistant) -> None:
+    """Perform operations after a backup finishes."""
+    instance: Recorder = hass.data[DATA_INSTANCE]
+    _LOGGER.info("Backup end notification, releasing write lock")
+    instance.unlock_database()
```

Comment:
These functions need to support returning a value indicating success or not:
- If `async_pre_backup` fails, backing up is not possible
- If `async_post_backup` fails, backing up needs to be restarted

Reply:
it support raising